pkg/kp/lock_test.go

```diff
@@ -0,0 +1,39 @@
+package kp
+
+import (
+	"testing"
+	"time"
+)
+
+const (
+	lockMessage = "Locked by lock_test.go"
+)
+
+func TestLock(t *testing.T) {
```

Comment:
I don't believe that locking does prevent reads or writes, we just use a convention that any process performing a replication must first acquire locks named a particular way.